src/main/resources/jbang-catalog.json

```diff
@@ -0,0 +1,29 @@
+{
+  "templates": {
+    "agent": {
+      "file-refs": ["init-agent.java.qute"],
```

Comment:
I suggest these should be key/values to capture their intended path, i.e.:

```json
"{baseName}.java": "init-agent.java.qute"
"readme.md": "readme.md.qute"
"resources/application.properties": "app.props"
```

Reply:
Yes, we can certainly do that. Right now they use the same format as //FILES ("ref=destination")

Reply:
Perhaps my main issue with this format is that it will cause a lot of duplicates in many situations:

```
"file1.java": "file1.java",
"file2.java": "file2.java",
"file3.java": "file3.java",
"file4.java": "file4.java",
```

I also don't know if we should force `.qute` files to specify their target name or just automatically strip the extension?

Reply:
Perhaps a "shortcut format" could be something like:

```
"{baseName}.java": "init-agent.java.qute",
"readme.md.qute": "",
"resources/application.properties": "app.props",
"file1.java": "",
"file2.java": "",
"file3.java": "",
"file4.java": "",
```

Where an empty string would mean "use the same name", but it would also use auto-renaming in the case of special files like the `.qute` ones, where `foo.java.qute` would become `foo.java`.

Wdyt?

Reply:
NB: An in-between option could be to do that auto-renaming at the moment of running `template add` only. That makes the contents of the catalog.json file a little less "magicky" perhaps:

```
"{baseName}.java": "init-agent.java.qute",
"readme.md", "readme.md.qute": "",
"resources/application.properties": "app.props",
"file1.java": "",
"file2.java": "",
"file3.java": "",
"file4.java": "",
```

Reply:
i'm not sure how required/useful is to shorten these as they are not going to be "big" ...and in just a few files will have the case of key/path == templatename/path.

but yes: 

```
"{baseName}.java": "init-agent.java.qute",
"readme.md": "readme.md.qute",
"resources/application.properties": "app.props",
"file1.java": "",
"file2.java": "",
"file3.java": "",
"file4.java": "",
``` 

looks right to me. note, I changed the readme.md to be what I belive it should look like (no commas in the key).